android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/Utils.java

```diff
@@ -241,6 +273,11 @@ public static void setPwsEndpointPreference(Context context, String endpoint) {
   public static void setSharedPreferencesDefaultValues(Context context) {
     PreferenceManager.setDefaultValues(context, R.xml.settings, false);
     setPwsEndpointPreference(context, getCurrentPwsEndpointString(context));
+    if (restoreMainPrefSettings(context)) {
+      setOptInPreference(context);
+      deleteFile(context, MAIN_PREFS_KEY);
```

Comment:
The `MAIN_PREFS_KEY` refers to the opt in and the `DISCOVERY_SERVICE_PREFS_KEY` refers to the cache.